src/handlers/YT.ts

```diff
@@ -1,105 +1,117 @@
-import { AccountSnapshot } from "../schema/schema.ts";
-import {
-  PendleYieldTokenContext,
-  RedeemInterestEvent,
-  TransferEvent,
-} from "../types/eth/pendleyieldtoken.js";
 import { updatePoints } from "../points/point-manager.js";
-import { MISC_CONSTS } from "../consts.js";
-import {
-  getUnixTimestamp,
-  isPendleAddress,
-  getAllAddresses,
-} from "../helper.js";
-import { readAllUserERC20Balances, readAllYTPositions } from "../multicall.js";
+
+import { AccountSnapshotYT, RerunSnapshot } from "../schema/schema.ts";
+
+import { PendleYieldTokenContext } from "../types/eth/pendleyieldtoken.js";
+
+import { getUnixTimestamp, getAllYTAddresses } from "../helper.js";
+
 import { EVENT_USER_SHARE, POINT_SOURCE_YT } from "../types.js";
 
-/**
- * @dev 1 YT EZETH is entitled to yields and points generated by 1 ETH worth of EZETH (0.997... something)
- */
+import { PENDLE_POOL_ADDRESSES, MISC_CONSTS } from "../consts.js";
 
-export async function handleYTTransfer(
-  evt: TransferEvent,
-  ctx: PendleYieldTokenContext
-) {
-  await processAllYTAccounts(
-    ctx,
-    [evt.args.from.toLowerCase(), evt.args.to.toLowerCase()],
-    false
-  );
-}
+import { readAllUserERC20Balances, readAllYTPositions } from "../multicall.js";
 
-export async function handleYTRedeemInterest(
-  evt: RedeemInterestEvent,
-  ctx: PendleYieldTokenContext
-) {
-  await processAllYTAccounts(ctx, [evt.args.user.toLowerCase()], false);
-}
+const RERUN_KEY = `RERUN:${POINT_SOURCE_YT}`;
 
-export async function processAllYTAccounts(
+export async function processYTAccounts(
   ctx: PendleYieldTokenContext,
-  addressesToAdd: string[] = [],
-  shouldIncludeDb: boolean = true
+  addressesToAdd: string[] = []
 ) {
-  if (await ctx.contract.isExpired()) {
-    return;
-  }
+  let timestamp = BigInt(getUnixTimestamp(ctx.timestamp));
+  let rerunSnapshot = await ctx.store.get(RerunSnapshot, RERUN_KEY);
+  if (!rerunSnapshot)
+    rerunSnapshot = new RerunSnapshot({
+      id: RERUN_KEY,
+      ended: false,
+      updatedAt: timestamp,
+    });
 
-  const allAddresses = shouldIncludeDb ? await getAllAddresses(ctx) : [];
-  for (let address of addressesToAdd) {
-    address = address.toLowerCase();
-    if (!allAddresses.includes(address) && !isPendleAddress(address)) {
-      allAddresses.push(address);
+  if (rerunSnapshot.ended) return;
+
+  const addressesSet: Set<string> = new Set<string>();
```

Comment:
This is a proposal spreading multiple lines, and for optimization purpose, could save O(N) operation, (( not sure if it could make impact tho...:
from array to set and set to array the time complexity is O(N) I think we could optimize here by using array only and not switching between the set and list

** the following suggestion is just pseudo code
```suggestion
let addressToProcess = [];
for (address in addressToAdd){
  if(!isPendleAddress(address)){
    addressToProcess.push(address)
  }
}
```

Reply:
yeah but we have to check if the address is not added twice
which needs more processing